pybamm/solvers/processed_variable.py

```diff
@@ -181,10 +209,21 @@ def initialise_1D(self):
 
         # set up interpolation
         # note that the order of 't' and 'space' is the reverse of what you'd expect
+        # TODO: fix processing when variable is a function of space only
+        if fixed_t:
+            # Hack for 1D space only
 
-        self._interpolation_function = interp.interp2d(
-            self.t_sol, space, entries_for_interp, kind="linear", fill_value=np.nan
-        )
+            def interp_fun(t, z):
+                out = interp.interp1d(
+                    space, entries_for_interp[:, 0], kind="linear", fill_value=np.nan
+                )(z)
```

Comment:
`interp.interp1d( space, entries_for_interp[:, 0], kind="linear", fill_value=np.nan)` should be outside the function so that it's not recomputed each time

Reply:
thanks!